OrbitGl/ProcessManager.cpp

```diff
@@ -55,7 +58,26 @@ void ProcessManagerImpl::SetProcessListUpdateListener(
   process_list_update_listener_ = listener;
 }
 
-std::vector<ProcessInfo> ProcessManagerImpl::GetProcessList() const {
+outcome::result<std::vector<ModuleInfo>, std::string>
+ProcessManagerImpl::GetModuleList(uint32_t pid) {
+  GetModuleListRequest request;
+  GetModuleListResponse response;
+  request.set_process_id(pid);
+
+  std::unique_ptr<grpc::ClientContext> context = CreateContext();
+  grpc::Status status =
+      process_service_->GetModuleList(context.get(), request, &response);
+
+  if (!status.ok()) {
+    return status.error_message();
```

Comment:
I assume that the outcome library understands this is a failure based on the type, but I think it would be clearer to wrap this with `outcome::failure` nonetheless.